packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,107 @@
+import { TSESTree } from "@typescript-eslint/typescript-estree";
+import {
+  Reference,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+
+export = {
+  name: "load-object-before-read",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      loadBeforeRead: "An explicit load call on '{{name}}' for '{{loadValue}}' needs to be made before reading a proxy object",
+    },
+    docs: {
+      description: 
+        "Before you can read the properties of a proxy object, you must explicitly load the properties",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Possible Errors",
+      recommended: <false | "error" | "warn">false,
+      url: 
+        "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function getValueThatHadToBeLoaded(referenceNode: Reference): string | undefined {
+      if(referenceNode.identifier.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && referenceNode.identifier.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return referenceNode.identifier.parent.property.name;
+      }
+      return undefined;
+    }
+
+    function isLoadFunction(reference: Reference): boolean {
+      if(reference.identifier.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && reference.identifier.parent.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && reference.identifier.parent.property.name === "load") {
+        return true;
+      }
+      return false;
+    }
+
+    function wasCreatedByGetFunction(referenceNode: Reference): boolean {
+      const variable = referenceNode.resolved;
```

Comment:
Should pass in the actual node you are testing instead of an indirection to it (makes it more generic).

I also think the name should be a little clearer ("what are you testing was created by a get function).